modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java

```diff
@@ -730,19 +731,55 @@ private Map<String, Map<String, String>> _getActions(
 
 		return HashMapBuilder.<String, Map<String, String>>put(
 			"delete",
-			addAction(
-				"UPDATE", cpDefinition.getCPDefinitionId(), "deleteProduct",
-				_cpDefinitionModelResourcePermission)
+			() -> {
+				Map<String, String> stringStringMap = addAction(
+					"UPDATE", cpDefinition.getCPDefinitionId(), "deleteProduct",
+					_cpDefinitionModelResourcePermission);
+
+				String href = stringStringMap.get("href");
+
+				stringStringMap.put(
+					"href",
+					StringUtil.replace(
+						href, String.valueOf(cpDefinition.getCPDefinitionId()),
+						String.valueOf(cpDefinition.getCProductId())));
+
+				return stringStringMap;
```

Comment:
Map<String ,String> action is fine. Once you refactor, you can send this directly to liferay-commerce.